src/codegen/c.zig

```diff
@@ -7604,12 +7604,17 @@ fn writeMemoryOrder(w: anytype, order: std.builtin.AtomicOrder) !void {
     return w.writeAll(toMemoryOrder(order));
 }
 
-fn toCallingConvention(call_conv: std.builtin.CallingConvention) ?[]const u8 {
-    return switch (call_conv) {
-        .Stdcall => "stdcall",
-        .Fastcall => "fastcall",
-        .Vectorcall => "vectorcall",
-        else => null,
+fn toCallingConvention(cc: std.builtin.CallingConvention, zcu: *Zcu) ?[]const u8 {
+    return switch (cc) {
+        .auto, .naked => null,
+        .x86_stdcall => "stdcall",
+        .x86_fastcall => "fastcall",
+        .x86_vectorcall, .x86_64_vectorcall => "vectorcall",
+        else => {
+            // `Zcu.callconvSupported` means this must be the C callconv.
+            assert(cc.eql(zcu.getTarget().defaultCCallingConvention().?));
+            return null;
+        },
```

Comment:
FWIW, many of the new CCs can be added here. I did an early attempt at this here: https://github.com/alexrp/zig/commit/da280efc88be17237d357473e5a597d76f37747c#diff-30db613bb202b1efd3f74ea57ce8c58c24679118e5a6cc3c90afe1a3e1f174e2L7495-L7502 (incomplete)

Can also just be a follow-up PR of course.

Reply:
Oh that's great, I'll take a look at that commit and potentially add it into this PR. Thanks!